### ISSUE TITLE
Sort similar users by interest

### ISSUE THREAD

**Issue by JediMaster25** [nicotine-plus/nicotine-plus#2096]
### Describe the solution you'd like
Right now the interest feature doesn't seem to work. I've tried disliking every tag related to music and most similar users had only music. Maybe the users could be sorted by interest instead of alphabetically by default.

**Comment by mathiascode** [comment]
It seems like the similar users feature only takes likes into account, while dislikes affect the global item recommendations. Unfortunately we don't control the results provided by the interests feature, the Soulseek server uses some private algorithm to return results. The sorting order returned by the server feels arbitrary.

**Comment by slook** [comment]
It is a mysterious lottery, the current UX does not allow the feature to be as useful as it could be.

In the past there was the ability to select a user as the item in order to find similar users to that user, but unfortunately N+'s current interface does not allow this anymore.

**Comment by mathiascode** [comment]
> In the past there was the ability to select a user as the item in order to find similar users to that user, but unfortunately N+'s current interface does not allow this anymore. Such functionality might be desirable from elsewhere in the program, such as in the User context menu.

Are you sure? I don't recall this feature, and the Soulseek protocol doesn't contain functionality to do this.

**Comment by slook** [comment]
Yes, you used to be able to double-click on a User in the Similar Users list and this triggered an update of the list.

I assumed that the Username was used for the Thing parameter of the interest item request.

**Comment by mathiascode** [comment]
> You used to be able to double-click on a User in the Similar Users list and this triggered an update of the list (presumably to list users similar to the clicked-on User). Nowadays, the default action is Send Private Message to User instead.

The double-click action has always been private messages. https://github.com/nicotine-plus/nicotine-plus/blob/b8fa28b5dd27eafd44f2734ea9cb495a3cdf2e07/pynicotine/gtkgui/frame.py#L1096-L1100

**Comment by slook** [comment]
Maybe I was mistaken about double-click being used for this, perhaps it was one of the global/local Recommendation buttons then.

Here it looks like `thing` can be a Username:

https://github.com/nicotine-plus/nicotine-plus/blob/b8fa28b5dd27eafd44f2734ea9cb495a3cdf2e07/pynicotine/gtkgui/frame.py#L1184-L1187
